lib/internal/modules/esm/get_format.js

```diff
@@ -66,6 +72,16 @@ function extname(url) {
   return '';
 }
 
+/**
+ * Determine whether the given file URL is under a `node_modules` folder.
+ * This function assumes that the input has already been verified to be a `file:` URL,
+ * and is a file rather than a folder.
+ * @param {URL} url
+ */
+function underNodeModules(url) {
+  return StringPrototypeIndexOf(url.pathname, '/node_modules/') !== -1;
```

Comment:
this seems like it would incorrectly return `true` for a URL that contains `/node_modules/../` etc?

also, currently, if i have a node application that *happens* to be contained inside a node_modules folder somewhere, I don't believe there's currently any behavior change as a result of that - this seems like it would make the new flag a noop in that case?

Reply:
The URL at this position would be fully normalized, so that `..` segments cannot appear. I posted before reading your comments, but address the case you just brought up in https://github.com/nodejs/node/pull/49869#discussion_r1336530325.